pkgs/by-name/bp/bpftrace/package.nix

```diff
@@ -41,9 +41,21 @@ stdenv.mkDerivation rec {
     "-DBUILD_TESTING=FALSE"
     "-DLIBBCC_INCLUDE_DIRS=${bcc}/include"
     "-DINSTALL_TOOL_DOCS=OFF"
-    "-DUSE_SYSTEM_BPF_BCC=ON"
+    "-DSYSTEM_INCLUDE_PATHS=${glibc.dev}/include"
   ];
 
+  patches = [
+    # https://github.com/bpftrace/bpftrace/pull/3243 (merged)
+    ./override-system-headers.patch
+    # https://github.com/bpftrace/bpftrace/pull/3152 (merged)
+    ./tcp-bt-no-includes.patch
+    # https://github.com/bpftrace/bpftrace/pull/3262 (merged)
+    ./runqlat-bt-no-includes.patch
+    # https://github.com/bpftrace/bpftrace/pull/3242 (merged)
+    ./kheaders-not-found-message-fix.patch
+    # https://github.com/bpftrace/bpftrace/pull/3265
+    ./kheaders-not-found-message-only-on-error.patch
```

Comment:
Why are we not using fetchpatch to obtain them?

Reply:
>  The problem is that they ran clang format between 0.20 and 0.21 so none of the patches apply directly and I had to make it files specific to 0.20...

Reply:
(I'm switching to fetchpatch for the couple of patches that stay in https://github.com/NixOS/nixpkgs/commit/875e88e8c28ff4cb4ab9065fe01291249ce35700 for the bpftrace 0.21.1 upgrade -- but thought it'd make sense to only fix things so we can backport that to stable branch and only update for master; if it's less work we can just update everything... But the current situation is broken so it'd be great to merge something, whichever it is)

Reply:
Good to know, then we can obviously not fetch them. Just re-request a review from me when you are done.

Reply:
I consider this PR done -- the way I see it the last bpftrace 0.20.4 had regressions due to no longer extracting ikheaders, so:
- this PR fixes just the regressions (the last upstream PR has been merged since so it's all done)
- once that PR is in we can
  - backport exactly this PR to 24.05 so the regressions get fixed there
  - update nixpkgs-unstable to bpftrace 0.21.1, including using patches as per upstream

If we consider that keeping 24.05 to the older version isn't worth it, then we can just skip this PR and update straight to bpftrace 0.21.1 here then backport that update to 24.05; I don't use 24.05 so I don't have a preference either way (I think the regressions ought to be fixed, I don't care if we fix it by backport to the freezed version or if we just update it)

... Writing this I realized we can also just retarget this PR to 24.05 and open a new one that just updates master, there's no real benefit of 24.05 work being a backport is there.

Anyway, all the dev is done (including the update to bpftrace 0.21.1 in my tree), just tell me what you'd like to merge :)

Reply:
@martinetd updating to 0.21.1 on stable sounds good to me.

Reply:
@Mic92 Thanks!

I've opened https://github.com/NixOS/nixpkgs/pull/327053 to update master to 0.21.1.

Okay for the update to stable, I don't have a preference either way; will backport that new PR to stable directly then